frontend/src/components/cesium/CesiumMap.jsx

```diff
@@ -188,12 +185,37 @@ const CesiumMap = ({onLocationSelect, id, setDroneLocation}) => {
           }}
         />
       ))}
+      {safeZones.map((zone, index) => (
+        <React.Fragment key={index}>
+          <Entity
+            position={zone.position}
+            billboard={{
+              image: zone.image,
+              scale: 0.15,
+              verticalOrigin: VerticalOrigin.BOTTOM,
+              color: Color.WHITE
+            }}
+          />
+          {zone.radius > 0 && (
+            <Entity
+              position={zone.position}
+              ellipse={{
```

Comment:
@syedomair0 You can move this ellipsoidal entity and also the pin dropping logic at newline 117 into a whole new component. Make sure to pull the latest from frontend-updates before updating this as the old code from CesiumMap component was split into multiple smaller components.

Reply:
i am doing refactoring on a separate branch which can include this cleanup as well. so is it okay if i do this in a different PR ?

Reply:
@syedomair0 This PR cannot be considered for merging because the target branch was wrong. It's supposed to be `frontend-updates`. Please move your old work (along with addressing the review comments from this PR) into a new branch, and then create a new PR